recommender/engine/rl_agent/preprocessing/filters_embedder.py

```diff
@@ -1,17 +1,92 @@
-from typing import List
+# pylint: disable=no-member, too-few-public-methods
+
+"""Implementation of the Filters Embedder"""
+
+from typing import Dict, Any
 
 import torch
+from pandas import DataFrame
+
+from recommender.engine.pre_agent.models import load_last_module, SERVICES_AUTOENCODER
+from recommender.engine.pre_agent.preprocessing import load_last_transformer, SERVICES
+from recommender.engine.pre_agent.preprocessing.preprocessing import SERVICE_COLUMNS
+from recommender.models import (
+    Category,
+    Provider,
+    Platform,
+    ScientificDomain,
+    TargetUser,
+)
+
+
+def filters_to_df(search_data: Dict[str, Any]) -> DataFrame:
+    """
+    Transform filters from search_data dict into service-like Pandas dataframe.
+
+    Args:
+        search_data: Search data with filters.
+
+    Returns:
+        Service-like pandas dataframe.
+
+    """
 
-from models import SearchData
+    dataframe = DataFrame(columns=SERVICE_COLUMNS)
+
+    countries = search_data.get("geographical_availabilities")
+    categories = Category.objects(id__in=search_data.get("categories"))
+    providers = Provider.objects(id__in=search_data.get("providers"))
+    platforms = Platform.objects(id__in=search_data.get("related_platforms"))
+    scientific_domains = ScientificDomain.objects(
+        id__in=search_data.get("scientific_domains")
+    )
+    target_users = TargetUser.objects(id__in=search_data.get("target_users"))
+
+    row_dict = {
+        "countries": list(countries),
+        "categories": [category.name for category in categories],
+        "providers": [provider.name for provider in providers],
+        "resource_organisation": [],
+        "platforms": [platform.name for platform in platforms],
+        "scientific_domains": [sci_dom.name for sci_dom in scientific_domains],
+        "target_users": [(tu.name, tu.description) for tu in target_users],
+        "access_modes": [],
+        "access_types": [],
+        "trls": [],
+        "life_cycle_statuses": [],
+    }
+
+    dataframe = dataframe.append(row_dict, ignore_index=True)
+
+    return dataframe
 
 
 class FiltersEmbedder:
+    """Embedder of filters"""
+
     def __init__(self):
```

Comment:
I suggest we pass services_embedder and service_transformer as a variable here (as stated in the architecture convention) as it will be used by many modules in RLAgent.

Reply:
Sure, good point.